### DISCUSSION TITLE
Why is popup render limited to raw HTML?

### DISCUSSION THREAD

**Discussion by andrewigk** [maplibre/maplibre-gl-geocoder#194] [Q&A]
In the documentation, it says:
popupRender
A function that specifies how the results should be rendered in the popup menu. Any HTML in the returned string will be rendered.

(item): string
Parameters
item: [CarmenGeojsonFeature](https://maplibre.org/maplibre-gl-geocoder/types/CarmenGeojsonFeature.html)
Returns string

In my use case where I'm using React, I haven't found a good solution to creating pop-ups that have more than just static text populating them. For example, if you wanted to include a button, there doesn't seem to be a way to pass a function to the onClick property of the button in raw HTML.

**Comment by HarelM** [comment]
This is an implementation that doesn't need any framework or other lib dependencies, so it can't render a react popup.
Having said that, you should be able to listen to the relevant event and draw the popup using some react maplibre wrapper I suppose, can't you?
